spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/SparkUtils.java

```diff
@@ -84,7 +85,7 @@ static void validateConfigParameters(Schema inputSchema, @Nullable String featur
       throw new IllegalArgumentException("Cannot specify values for both featuresToInclude and featuresToExclude. " +
                                            "Please specify fields for one.");
     }
-    Map<String, Integer> fields = new HashMap<>();
+    Map<String, Integer> fields = new LinkedHashMap<>();
```

Comment:
The SparkUtils class is being used by the other spark ml plugins and the order error will occur for every plugin. Also, the sequence of fields in the output schema created using the `getFeatureList()` will differ for different reducers for the current implementation.
The test `testIncludeAllFeatures()` checks for a scenario where the `featuresToInclude` and `featuresToInclude` parameters are both empty and hence all the fields will be checked for the data types in the sequence they appear in the input schema. Not sure if changing the test case will solve the basic issue.
Please suggest.